hs_modelinstance/models.py

```diff
@@ -58,20 +58,50 @@ def modelProgramIdentifier(self):
 
     @classmethod
     def create(cls, **kwargs):
-        shortid = kwargs['model_name']
-        # get the MP object that matches.  Returns None if nothing is found
-        obj = ModelProgramResource.objects.filter(short_id=shortid).first()
-        metadata_obj = kwargs['content_object']
-        title = obj.title
-        return super(ExecutedBy,cls).create(model_program_fk=obj, model_name=title,content_object=metadata_obj)
+
+        # when creating a new resource version, we need to lookup the existing model_program_fk
+        if 'model_program_fk' in kwargs:
```

Comment:
What is the value of model_program_fk? is it a resource_id of a hydroshare resource or some other internal key of this table in django db?